Dockerfile

```diff
@@ -28,7 +27,11 @@ RUN cp /tmp/requirements.txt /tmp/req.bak \
 # TODO: remark for issue [revise the dockerfile #1276](https://github.com/Project-MONAI/MONAI/issues/1276)
 # please specify exact files and folders to be copied -- else, basically always, the Docker build process cannot cache
 # this or anything below it and always will build from at most here; one file change leads to no caching from here on...
-COPY . .
+
+COPY LICENSE setup.py setup.cfg versioneer.py runtests.sh ./
+COPY tests ./tests
+COPY monai ./monai
+COPY .git ./.git
```

Comment:
perhaps we could have`.gitignore`, `.gitattributes` as well?

Reply:
I can see .gitattributes interacts with versioneer.py, so it's necessary to add it back.  Additionally, MANIFEST.in is used during python setup.py and I put it back to Dockerfile and removed it from .dockerignore.
I am not 100% sure if .gitignore is used inside container.  However, its content should be pretty stable and won't invalidate cache often, so it's also included, too.
The updated Dockerfile and .dockerignore was just force-pushed.